### PR TITLE
fix: meeting link is not updated and broken ui in edit location dialog

### LINKED ISSUE #15497: (Very annoying) Guest doesn't see updated meeting link on booking page
Found a bug? Please fill out the sections below. 👍

### Issue Summary

When you change the link of a meeting the meeting page still shows attendees link.

### Steps to Reproduce

1. Set a custom link.
image

2. Someone books a meeting
3. You add the actual link to the booked meeting and click update
image
4. The guest visits the booking page and sees the wrong link
image

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

This is a bug because even if the actual meeting link is in the cal invite people for some reason open the booking page often and they see the wrong meeting link.

### Actual Results

- Ivitee sees the wrong link

### Expected Results

- They see the new updated link

### Technical details

- I'm using the latest hosted version

### Evidence

- How was this tested? Steps to reproduce make it pretty clear but I could add a video if anyone else is having trouble with this.

### PR DESCRIPTION
**Pull request by anikdhabal**
## What does this PR do?

- Fixes #15497 
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

before:-
Screenshot 2024-06-20 091049

Now:-
Screenshot 2024-06-20 090909

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected)
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com)
- [ ] I have added or modified automated tests that prove my fix is effective or that my feature works (PRs might be rejected if logical changes are not properly tested)

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

File: packages/trpc/server/routers/viewer/bookings/editLocation.handler.ts [hunk 1]
```diff
@@ -118,6 +118,10 @@ export const editLocationHandler = async ({ ctx, input }: EditLocationOptions) =
         },
         data: {
           location,
+          metadata: {
+            ...(typeof booking.metadata === "object" ? booking.metadata : {}),
+            videoCallUrl: location,
```
Comment (supalarry) [inline packages/trpc/server/routers/viewer/bookings/editLocation.handler.ts:123]: What if location is a phone number or address? We then would store it as videoCallUrl in metadata.

Comment (supalarry) [reply packages/trpc/server/routers/viewer/bookings/editLocation.handler.ts:123]: Here is updated booking metadata:

Screenshot 2024-06-21 at 15 05 27

I think we need to update `videoCallUrl` only if its a link or cal daily video.

**Review by supalarry — DISMISSED** [review]
Thanks for the PR Anik. I request changes because of comment below.

**Comment by supalarry** [comment]
What I also found is that if one changes location of booking to cal video:

Screenshot 2024-06-21 at 15 02 19

saves and refreshes page and checks location, its still the previous one:
Screenshot 2024-06-21 at 15 02 33

File: packages/trpc/server/routers/viewer/bookings/editLocation.handler.ts (hunk 1 above)
Comment (supalarry) [reply packages/trpc/server/routers/viewer/bookings/editLocation.handler.ts:123]: however, if the location changes from link to phone, then we need to clear metadata.

**Comment by Udit-takkar** [comment]
@anikdhabal any update on this PR?

File: packages/trpc/server/routers/viewer/bookings/editLocation.handler.ts (hunk 1 above)
Comment (anikdhabal) [reply packages/trpc/server/routers/viewer/bookings/editLocation.handler.ts]: Fixed it. Actually, when the location is a link or a static video app, we only set the videoCallUrl inside the metadata.

**Comment by anikdhabal** [comment]
> What I also found is that if one changes location of booking to cal video:

Actually, we faced this error in dev mode if the Cal video API key is null. In production, I’ve tested it and it works fine

**Comment by anikdhabal** [comment]
Cc @supalarry

**Review by PeerRich — DISMISSED** [review]

**Review by Udit-takkar — COMMENTED** [review]
Text area min and max characters text > Create a new event E2E test is failing because it's not able to find any slot at the end of the month. so we can fix it by going to next month if there is no available slots.

**Review by Udit-takkar — APPROVED** [review]

**Comment by RayBB** [comment]
Thank goodness this is finally fixed! Much appreciated.

**Comment by zomars** [comment]
@Udit-takkar @PeerRich this PR shouldn't have been merged yet. Will revert. Unless I'm missing more context if we have a location that doesn't start with "https" It will wipe out all existing booking metadata.

image

File: packages/trpc/server/routers/viewer/bookings/editLocation.handler.ts [hunk 2]
```diff
@@ -118,6 +118,12 @@ export const editLocationHandler = async ({ ctx, input }: EditLocationOptions) =
         },
         data: {
           location,
+          metadata: location.startsWith("http")
+            ? {
+                ...(typeof booking.metadata === "object" ? booking.metadata : {}),
+                videoCallUrl: location,
+              }
+            : {},
```
Comment (zomars) [inline packages/trpc/server/routers/viewer/bookings/editLocation.handler.ts:126]: This condition will wipe out al previous booking metadata which I don't think is desirable.

Comment (anikdhabal) [reply packages/trpc/server/routers/viewer/bookings/editLocation.handler.ts:126]: Add a suggestion https://github.com/calcom/cal.com/pull/16011#issuecomment-2261267087

**Pull request merged**